### PR TITLE
Keyboard shortcuts

### PR DESCRIPTION
**Pull request by P-man2976**
- Added Keyboard shortcuts. (mainly based on Spotify Keyboard shortcuts)
  - using `react-hotkeys-hook`  library

|Key|Action|Note|
|---|---|---|
|`Ctrl+n` `Cmd+n`|Open "Create New Playlist" modal|Only working on iOS PWA etc (usually conflicts with browser)|
|`Space`|Play / Pause||
|`Ctrl+r` `Cmd+r`|Change repeat mode||
|`Ctrl+s` `Cmd+s`|Toggle shuffle mode||
|`Ctrl+(left/right)` `Cmd+(left+right)`|Previous / Next track||
|`Ctrl+(up/down)` `Cmd+(up/down)`|Up / Down volume|Every 5%|
|`Ctrl+Shift+(up/down)` `Cmd+Shift+(up/down)`|Unmute or Max volume / Mute|If muted, `Ctrl+Shift+up` restores previous volume|
|`Ctrl+l` `Cmd+l`|Give focus to search field||
|`Ctrl+p` `Cmd+,`|Open settings page||
|`Ctrl+Shift+w` `Cmd+Shift+w`|Log out|Only working on iOS PWA etc|
|`f` / `esc`|Open / Close Musicdex full-screen player|Based on YouTube shortcut|

### CODE REVIEW

File: src/components/player/YoutubePlayer.tsx [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function usePlayer(player?: YouTubePlayer) {
           setVolume(await player.getVolume());
           setMuted(await player.isMuted());
         }
-      }, 333);
+      }, 200);
```
Comment (RiceCakess) [inline src/components/player/YoutubePlayer.tsx:73]: Is this needed change?

**Review by RiceCakess — APPROVED** [review]

File: src/components/player/YoutubePlayer.tsx (hunk 1 above)
Comment (P-man2976) [reply src/components/player/YoutubePlayer.tsx:73]: Volume change event from keyboard shortcut is limited to the update interval of the `YoutubePlayer` component.
So, I thought update interval should be accelerated.
(but yeah it's a logic problem. I'll think better solution.)

Comment (RiceCakess) [reply src/components/player/YoutubePlayer.tsx:73]: maybe if you change it directly using player, without waiting for react states to update. Something like

```js
player.setVolume(await player.getVolume() + 5)
```

The visual will lag behind by 333ms, but that's ok

Comment (P-man2976) [reply src/components/player/YoutubePlayer.tsx:73]: Thanks for the advice. I'll change to it.

**Comment by RiceCakess** [comment]
Nice work! Thanks!

**Pull request merged**